runtime/sema/check_composite_declaration.go

```diff
@@ -1963,6 +1995,111 @@ func (checker *Checker) enumMembersAndOrigins(
 	return
 }
 
+func (checker *Checker) checkDefaultDestroyParamExpressionKind(
+	arg ast.Expression,
+) {
+	switch arg := arg.(type) {
+	case *ast.StringExpression,
+		*ast.BoolExpression,
+		*ast.NilExpression,
+		*ast.IntegerExpression,
+		*ast.FixedPointExpression,
+		*ast.PathExpression:
+		break
+	case *ast.IdentifierExpression:
+		// these are guaranteed to exist at time of destruction, so we allow them
+		if arg.Identifier.Identifier == SelfIdentifier || arg.Identifier.Identifier == BaseIdentifier {
```

Comment:
Maybe refactor the repeated `arg.Identifier.Identifier` expression throughout the case into a variable to make the code easier to read